onebot/plugins/psa.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""
+================================================
+:mod:`onebot.plugins.psa` PSA
+================================================
+
+This plugin allows admins to send broadcasts
+"""
+
+import irc3
+from irc3 import plugin
+from irc3.plugins.command import command
+
+
+@plugin
+class PSAPlugin(object):
+    """PSA Plugin"""
+
+    requires = [
+        'irc3.plugins.command',
+    ]
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.channels = set()
+
+    @irc3.event(irc3.rfc.JOIN)
+    def register_channel(self, mask, channel, **kwargs):
```

Comment:
The channel list will grow infinitely, I'd rather not send a gazillion messages to channels the bot has already left. [This example](https://github.com/gawel/irc3/blob/master/irc3/plugins/userlist.py#L98) should be useful to keep the set mostly accurate (not capturing _all_ part events is fairly acceptable).